src/Razor/SourceGenerator/src/RazorSourceGenerationContext.cs

```diff
@@ -0,0 +1,183 @@
+            }
+
+            globalOptions.TryGetValue("build_property._RazorSourceGeneratorDebug", out var waitForDebugger);
+            globalOptions.TryGetValue("build_property._RazorSourceGeneratorWriteGeneratedOutput", out var writeOutput);
+
+            var razorConfiguration = RazorConfiguration.Create(razorLanguageVersion, configurationName, Enumerable.Empty<RazorExtension>());
+            var (razorFiles, cshtmlFiles) = GetRazorInputs(context);
+            var fileSystem = GetVirtualFileSystem(razorFiles, cshtmlFiles);
+
+            return new RazorSourceGenerationContext
+            {
+                RootNamespace = rootNamespace,
+                Configuration = razorConfiguration,
+                FileSystem = fileSystem,
+                RazorFiles = razorFiles,
+                CshtmlFiles = cshtmlFiles,
+                DesignTimeBuild = designTimeBuild == "true",
+                RefsTagHelperOutputCachePath = refsTagHelperOutputCachePath,
+                WaitForDebugger = waitForDebugger == "true",
+                WriteGeneratedContent = writeOutput == "true",
+            };
+        }
+
+        private static VirtualRazorProjectFileSystem GetVirtualFileSystem(IReadOnlyList<RazorInputItem> razorFiles, IReadOnlyList<RazorInputItem> cshtmlFiles)
+        {
+            var fileSystem = new VirtualRazorProjectFileSystem();
+            for (var i = 0; i < razorFiles.Count; i++)
+            {
+                var item = razorFiles[i];
+                fileSystem.Add(new SourceGeneratorProjectItem(
+                    basePath: "/",
+                    filePath: item.NormalizedPath,
+                    relativePhysicalPath: item.RelativePath,
+                    fileKind: FileKinds.Component,
+                    item.AdditionalText,
+                    cssScope: item.CssScope));
+            }
+
+            for (var i = 0; i < cshtmlFiles.Count; i++)
+            {
+                var item = cshtmlFiles[i];
+                fileSystem.Add(new SourceGeneratorProjectItem(
+                    basePath: "/",
+                    filePath: item.NormalizedPath,
+                    relativePhysicalPath: item.RelativePath,
+                    fileKind: FileKinds.Legacy,
+                    item.AdditionalText,
+                    cssScope: item.CssScope));
+            }
+
+            return fileSystem;
+        }
+
+        private static (IReadOnlyList<RazorInputItem> razorFiles, IReadOnlyList<RazorInputItem> cshtmlFiles) GetRazorInputs(GeneratorExecutionContext context)
+        {
+            List<RazorInputItem> razorFiles = null;
+            List<RazorInputItem> cshtmlFiles = null;
+
+            foreach (var item in context.AdditionalFiles)
+            {
+                var path = item.Path;
+                var isComponent = path.EndsWith(".razor", StringComparison.OrdinalIgnoreCase);
+                var isRazorView = !isComponent && path.EndsWith(".cshtml", StringComparison.OrdinalIgnoreCase);
```

Comment:
"!isComponent" is redundant check now. if path.EndsWith(".cshtml", StringComparison.OrdinalIgnoreCase) == true then path.EndsWith(".razor", StringComparison.OrdinalIgnoreCase) = false.